File: springboot/src/main/java/com/velidjanov/mew/mewscaffolds/entity/Layer.java
package com.velidjanov.mew.mewscaffolds.entity;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;

@NoArgsConstructor @Getter @Setter @ToString
public class Layer {

    private Long id;

    @NonNull
    private String name;

    @NonNull
    private Double angle = 0.0d;

    private Integer fibers;

    @NonNull
    private Double width;

    @NonNull
    private Double height;

    @NonNull
    private Double distanceBetweenFibers;

    @NonNull
    private Boolean isSinusoidal;

    @NonNull
    private Double amplitude;

    @NonNull
    private Double phase;

    @NonNull
    private Double phaseShift;

    @NonNull
    private Double temperature;

    @NonNull
    private Double pressure;

    @NonNull
    private Double speed;

    @NonNull
    private Double loopSpeed;

    @NonNull
    private Double loopRadius;

    @NonNull
    private Double waitIn;

    @NonNull
    private Double waitOut;

    @NonNull
    private Double distanceZ;

    public Layer(@NonNull Double height, @NonNull Double distanceBetweenFibers) {
        this.height = height;
        this.distanceBetweenFibers = distanceBetweenFibers;
        this.fibers = (int) (this.height / this.distanceBetweenFibers);
        this.fibers += 1;
    }
}

File: angularclient/src/app/model/layer/layer.ts
import { MewData } from '../abstract/mew-data';
import { SettingsConfig } from 'src/app/config/settings-config/settings-config';

export class Layer extends MewData {
    
    fibers: number;
    angle: number;
    width: number;
    height: number;
    distanceBetweenFibers: number;

    speed: number;
    loopSpeed: number;
    loopRadius: number;
    waitIn: number;
    waitOut: number;
    zDistance: number;
    angleIncrement: number;
    distanceBetweenFibersIncrement: number;
    
    isSinusoidal: boolean;
    amplitude: number;
    phase: number;
    phaseShift: number;

    constructor(id: number, name: string, parent: MewData) {
        super(id, name, null, parent);
        this.angle = SettingsConfig.instance.defaultLayerAngle;
        this.width = SettingsConfig.instance.defaultLayerWidth;
        this.height = SettingsConfig.instance.defaultLayerHeight;
        this.distanceBetweenFibers = SettingsConfig.instance.defaultDistanceBetweenFibers;
        this.fibers = Math.floor(this.height / this.distanceBetweenFibers) + 1;

        this.speed = SettingsConfig.instance.defaultLayerSpeed;
        this.loopSpeed = SettingsConfig.instance.defaultLayerLoopSpeed;
        this.loopRadius = SettingsConfig.instance.defaultLayerLoopRadius;
        this.waitIn = SettingsConfig.instance.defaultLayerWaitIn;
        this.waitOut = SettingsConfig.instance.defaultLayerWaitOut;
        this.zDistance = SettingsConfig.instance.defaultLayerZDistance;
        this.angleIncrement = SettingsConfig.instance.defaultLayerAngleIncrement;
        this.distanceBetweenFibersIncrement = SettingsConfig.instance.defaultLayerDistanceBetweenFibersIncrement;

        this.isSinusoidal = SettingsConfig.instance.defaultLayerIsSinusoidal;
        this.amplitude = SettingsConfig.instance.defaultLayerAmplitude;
        this.phase = SettingsConfig.instance.defaultLayerPhase;
        this.phaseShift = SettingsConfig.instance.defaultLayerPhaseShift;
    }
}
